psi4/driver/procrouting/proc.py

```diff
@@ -5053,3 +5110,170 @@ def run_efp(name, **kwargs):
         core.set_variable('EFP TORQUE', torq)
 
     return ene['total']
+
+def run_sapt0d(name, **kwargs):
+    """ Function to call SAPT0-D, uses existing SAPT0 and dispersion routines """
+
+    # First, we need to run SAPT0 without the usual dispersion    
+    ## Call SAPT0
+    optstash = p4util.OptionsState(
```

Comment:
Could we build this similar to the fsapt where we simply modify the existing routine instead of duplicating it?

Reply:
Yes I think that would make more sense. Thoughts on moving these -D routines to a helper file, and maybe even combining them into one function? Only think stopping me from doing that initially was handling setting variables.

Reply:
I'm for refactoring the -d analysis to a helper routine/file and passing in a label and/or bool to help with the needed toggles. can either pass in the wfn or return a dict of qcvars to be set on the wfn back in the proc.py routine.